packages/contracts/contracts/payoutStrategy/MerklePayoutStrategy/MerklePayoutStrategyImplementation.sol

```diff
@@ -71,7 +91,48 @@ contract MerklePayoutStrategyImplementation is IPayoutStrategy, Initializable {
    *
    * @param encodedDistribution encoded distribution
    */
-  function payout(bytes[] memory encodedDistribution) external override isRoundContract payable {
-    // TODO: Add payout logic
+  function batchPayout(Distribution[] calldata _distributions) external override isRoundContract payable { 
+
+    for (uint256 i = 0; i < _distributions.length; ++i) {
+        _distribute(_distributions[i]);
+    }
+
+    emit BatchPayoutTriggered(msg.sender);
+  }
+
+  function _distribute(Distribution calldata _distribution) internal {
```

Comment:
should this be public ?

Reply:
no - unless we want to support individual payouts